Error404 casi terminado.

JLeonN opened the pull request:
Solicité a **Christian** que creara una imagen para este componente; aún no estoy seguro si lo hará, pero mientras tanto, utilizaremos la que tenemos.

JLeonN committed:
Carpeta 404.

He creado una carpeta específica para almacenar el componente 404. Esto facilitará la organización, especialmente porque planeo agregar más contenido al error 404 en el futuro.

JLeonN committed:
Borrando Comentarios

😁😀

JLeonN committed:
Error 404 pronto?

El componente de Error 404 está casi terminado.
Solo falta agregarle algunos detalles más, pero me he quedado sin ideas por el momento.
Solicité a Christian que creara una imagen para este componente; aún no estoy seguro si lo hará, pero mientras tanto, utilizaremos la que tenemos.
Por otro lado, en VSC me aparece un error en el componente del botón relacionado con los PROPS. No sé la razón exacta, pero a pesar de eso, sigue funcionando correctamente.

JLeonN committed:
Error 404 terminado?

He terminado de pulir el error 404. Finalmente, opté por convertirlo en una tarjeta similar a la que muestra al 'Cowboy del día'

JLeonN committed:
Merge branch 'main' into Error404

JLeonN committed:
TarjetaDia con error

Aún persiste el problema en la tarjeta del día. Me vi obligado a agregar un condicional ('if') para validar la tarjeta. No estoy seguro si es la mejor solución, pero al menos ahora funciona. Sería conveniente revisar esta parte.

JLeonN committed:
errores ortográficos

Corrigiendo errores ortográficos y otros detalles adicionales.

JLeonN committed:
errores ortográficos

😁

JLeonN committed:
Error404 fin?

Hey Franco, necesito enviarte un mensaje de voz por WhatsApp o Discord, tengo un lío enorme. Más o menos he logrado resolverlo. Resulta que metí la pata con toda esta rama. La idea original era solo trabajar en el componente Error404, pero terminé modificando otras cosas para que ese componente quedara bien y, sin darme cuenta, terminé alterando casi todo, ¡ja! Bueno, luego te contacto por audio para explicarte mejor.

JLeonN committed:
Nuevo Inicio

He realizado modificaciones en la página de inicio. Ahora, esta sección mostrará únicamente dos tarjetas: una destacando al 'tonto del día' y otra al 'cowboy del día'. Toda la información adicional estará disponible en sus respectivas secciones, como la lista completa de 'tontos del día' en la sección dedicada a ello.

francosang commented on frontend/src/Componentes/Botones.jsx:
```diff
@@ -0,0 +1,33 @@
+import { Link } from "react-router-dom";
+
+const Botones = ({ button }) => {
+  if (!button) {
+    return;
+  }
```
Este fue lo que te dijo ChatGPT?

Creo que no es necesario, me parece que se puede borrar.
No sería correcto crear un boton que no tiene ruta ni informacion.

JLeonN replied:
Si eso mismo

JLeonN replied:
Acabo de eliminarlo y no tuve ningún problema, tal como me mencionaste. Sin embargo, en el componente de la tarjeta del día, todavía tengo dificultades.

JLeonN replied:
Franco, ya he solucionado el problema. Como mencionaste, un componente vacío puede generar un error. He comentado esa parte del código y eliminado el 'if', y ahora funciona correctamente.

francosang commented on frontend/src/Componentes/Pagina/InfoTonto/TarjetaDia.jsx:
```diff
@@ -1,9 +1,11 @@
+import "../EstilosElMasTonto.css";
 
-const TarjetaDia = () => {
+const TarjetaDia = ({ tarjeta }) => {
```
Este componente se usa tambien para mostrar errores, no solo para mostrar al tonto del dia o al cowboy del dia.

Que te parece ponerle un nombre mas generico?
Quizas que se llama `Tarjeta` sea mejor, no?

francosang replied:
Este componente se usa tambien para mostrar errores, no solo para mostrar al tonto del dia o al cowboy del dia.

Que te parece ponerle un nombre mas generico?
Quizas que se llama Tarjeta sea mejor, no?

francosang commented on frontend/src/Componentes/Pagina/Error404/Error404.jsx:
```diff
@@ -1,9 +1,28 @@
+    titulos: "¡Alto ahí, forastero!",
+    listaTitulos:
+      "Parece que has encontrado un territorio desconocido en el vasto desierto digital. No te preocupes, nuestros sheriff de la red están en camino para guiarte de regreso al pueblo principal. Mientras tanto, ¿por qué no te relajas con un poco de música en nuestro salon virtual?",
+    boton: "boton",
```
La propiedad `boton` no se usa.

francosang commented on frontend/src/Componentes/Pagina/Error404/Error404.jsx:
```diff
@@ -1,9 +1,28 @@
     <>
-      <h2>Error 404</h2> 
+      <div>
+        <TarjetaDia tarjeta={tarjeta} />
```
Creo que para el error 404 no tendriamos que usar el componente `Tarjeta`.

El componente `Tarjeta` tiene propiedades exclusivas para un tonto, como un nombre y titulos y una cantidad.
Creo que seria mejor que esta pagina no use este componente y tenga todo el HTML directamente.

JLeonN replied:
Al principio, estaba utilizando HTML plano, como mencionas. Después lo cambié, ¿por qué? Bueno, ¡ni yo mismo lo sé, jaja!

francosang commented on frontend/src/Componentes/Inicio.jsx:
```diff
@@ -0,0 +1,26 @@
+
+        {/* Tarjeta del cowboy del día */}
+        <h3 className="tituloH3">Cowboy del día de hoy es...</h3>
+        <TarjetaDia />
```
Es mejor no usar el componente tarjeta si no va a tener propiedades y va a estar vacio.
De esta forma te evitas poner el if dentro del componente.

Ademas, no es un uso correcto para el componente.

JLeonN replied:
La idea era incluir los datos del Cowboy del día, pero lamentablemente no pude completarlo a tiempo. Ese es el plan. Pretendo que funcione de manera similar al Tonto del día, pero enfocado en el Cowboy destacado del día.

JLeonN committed:
Borrando los - if.

Después de eliminar los 'if', todo está funcionando correctamente. Tuve que comentar el componente 'TarjetaDelDia', ya que por ahora no se está utilizando en 'Inicio'. Sin embargo, ¿qué sucederá cuando comience a usar el componente 'TarjetaDelDia' en 'Inicio'? ¿Será necesario volver a utilizar el 'if'?

JLeonN committed:
Nombre componente a Tarjeta

Cambio de nombre de componente a Tarjeta

JLeonN committed:
Error 404 terminado

He completado el Error 404! Espero que esta vez el componente esté listo pronto. Realicé varios cambios significativos, y debo admitir que me gusta más en su forma actual. De todos modos, si necesitas más ajustes, házmelo saber y seguimos trabajando en ello juntos.

JLeonN committed:
Borrando comentario

JLeonN committed:
hover Img 404

JLeonN committed:
btn hover

JLeonN committed:
Título más responsivo

francosang reviewed (approved):
Go go goooo!
